### PR TITLE
[loki-distributed] Handling the eks minor versions which has plus sign

### PR DESCRIPTION
**Pull request by bseenu**
Signed-off-by: Srinivas Boga <bseenu@gmail.com>

Below i tried to show the problem with a simple chart just having the configmap resource in its templates

```
bash-5.1$ cat capability/Chart.yaml
apiVersion: v2
name: capability
description: A Helm chart for testing kube version capability
version: 0.0.1


bash-5.1$ cat capability/templates/cm.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: capability
data:
  stringVersion: {{ .Capabilities.KubeVersion.Minor | quote }}
  intVersion: {{ .Capabilities.KubeVersion.Minor | int | quote }}


bash-5.1$ helm upgrade --install capability ./capability
Release "capability" does not exist. Installing it now.
NAME: capability
LAST DEPLOYED: Tue Nov  1 19:37:02 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None


bash-5.1$ kubectl get cm capability -o yaml
apiVersion: v1
data:
  intVersion: "0"
  stringVersion: 21+
kind: ConfigMap
metadata:
  annotations:
    meta.helm.sh/release-name: capability
    meta.helm.sh/release-namespace: default
  creationTimestamp: "2022-11-02T02:37:02Z"
  labels:
    app.kubernetes.io/managed-by: Helm
  name: capability
  namespace: default
  resourceVersion: "197404"
  uid: 2365fc95-d171-440a-8c4e-0d4374dfed3f


bash-5.1$ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"5835544ca568b757a8ecae5c153f317e5736700e", GitTreeState:"clean", BuildDate:"2022-09-21T14:33:49Z", GoVersion:"go1.19.1", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.14-eks-6d3986b", GitCommit:"8877a3e28d597e1184c15e4b5d543d5dc36b083b", GitTreeState:"clean", BuildDate:"2022-07-20T22:05:32Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.25) and server (1.21) exceeds the supported minor version skew of +/-1
```

### CODE REVIEW

**Comment by zanhsieh** [comment]
@bseenu 
Thanks for contribution. Could you run helm-docs and update README.md please?

**Comment by bseenu** [comment]
> @bseenu Thanks for contribution. Could you run helm-docs and update README.md please?

Done, Thanks

File: charts/loki-distributed/templates/ingester/deployment-ingester.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ spec:
         {{- include "loki.ingesterSelectorLabels" . | nindent 8 }}
         app.kubernetes.io/part-of: memberlist
     spec:
-      {{- if ge (.Capabilities.KubeVersion.Minor|int) 19 }}
+      {{- if ge (int (trimSuffix "+" .Capabilities.KubeVersion.Minor)) 19 }}
```
[the remark this answers is not in the document]
Comment (z0rc) [reply charts/loki-distributed/templates/ingester/deployment-ingester.yaml:35]: A much readable version would be `{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}`. Notice `-0` suffix, that's what makes the difference.

[the remark this answers is not in the document]
Comment (bseenu) [reply charts/loki-distributed/templates/ingester/deployment-ingester.yaml]: Changed to `semverCompare` now, Thanks

[the remark this answers is not in the document]
Comment (ldemailly) [reply charts/loki-distributed/templates/ingester/deployment-ingester.yaml]: and it works? with that weird + in our case?

for my education what does the `-0` changes/means?

[the remark this answers is not in the document]
Comment (z0rc) [reply charts/loki-distributed/templates/ingester/deployment-ingester.yaml]: > for my education what does the `-0` changes/means?

`-0` or anything after `-` is a pre-release suffix https://semver.org/#spec-item-9. Unfortunately multiple Kubernetes vendors use it to denote their own platform version, and this is a workaround to make semver comparison to work as expected.

> and it works? with that weird + in our case?

In provided example `.Capabilities.KubeVersion.GitVersion` is `v1.21.14-eks-6d3986b`, there is no `+` symbol in it. And hopefully won't be in near future, I haven't seen any cloud vendor changing GitVersion to be incompatible with semver.

[the remark this answers is not in the document]
Comment (ldemailly) [reply charts/loki-distributed/templates/ingester/deployment-ingester.yaml]: that's what I thought (that it'd be pre style) - I wonder how it ends up turning to "21+" in the string

[the remark this answers is not in the document]
Comment (bseenu) [reply charts/loki-distributed/templates/ingester/deployment-ingester.yaml]: just adding what i found 

`SemVer comparisons using constraints without a prerelease comparator will skip prerelease versions. For example, >=1.2.3 will skip prereleases when looking at a list of releases while >=1.2.3-0 will evaluate and find prereleases.`

http://masterminds.github.io/sprig/semver.html

[the remark this answers is not in the document]
Comment (bseenu) [reply charts/loki-distributed/templates/ingester/deployment-ingester.yaml]: it only works when we pass `-0`

[the remark this answers is not in the document]
Comment (ldemailly) [reply charts/loki-distributed/templates/ingester/deployment-ingester.yaml]: thanks for having verified. doesn't quite make sense but oh well.

**Comment by bseenu** [comment]
@zanhsieh Can you please review and merge this

Thanks

**Comment by zanhsieh** [comment]
@bseenu 
Please bump the version in Chart.yaml and update README.md. Thanks.

**Comment by bseenu** [comment]
> @bseenu Please bump the version in Chart.yaml and update README.md. Thanks.

@zanhsieh Done, updated

Thanks

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**